src/persist-client/src/impl/state.rs

```diff
@@ -172,18 +188,24 @@ where
         )
     }
 
-    fn writer(&mut self, id: &WriterId) -> Result<&mut WriteCapability<T>, InvalidUsage> {
+    fn writer(&mut self, id: &WriterId) -> &mut WriteCapability<T> {
         self.writers
             .get_mut(id)
-            // TODO: It is more likely that the lease expired.
-            .ok_or_else(|| InvalidUsage(anyhow!("writer not registered: {}", id)))
+            // The only (tm) ways to hit this are (1) inventing a WriterId
+            // instead of getting it from Register or (2) if a lease expired.
+            // (1) is a gross mis-use and (2) isn't implemented yet, so it feels
+            // okay to leave this for followup work.
```

Comment:
If we know (2) is coming, shouldn't we leave the error so whoever uses this in the meantime has the opportunity to thread the error through properly from the staart?

Reply:
I think that the difference is that the interface of `PersistClient` & friends would be closer to what we expect it to look like long term!

Reply:
My thoughts are not fully formed, but I was thinking: If a lease can be re-acquired and we're in a state where we can keep serving ongoing things, that's something the client should do transparently. If it cannot be re-acquired, then there's nothing the user of the client can do to recover (that's a strong assumption, but idk 🤷). In both cases we wouldn't bubble up an error.